docs/build-tools/maven.md

```diff
@@ -0,0 +1,61 @@
+---
+id: maven
+title: Maven
+---
+
+Maven is one of the most common build tools in the JVM ecosystem and it also allows for using scala through the [scala-maven-plugin](https://davidb.github.io/scala-maven-plugin/usage.html).
+
+## Automatic installation
+
+The first time you open Metals in a new workspace it prompts you to import the
+build. Select "Import build" to start automatic installation.
+
+This will create all the needed bloop files, however there will be a warning, since the semanticDB plugin was not added yet in the automatic import. This will be added later. Most features should work without it, however some require semanticDB files to be provided alongside compiled data. To do that we need a couple of steps that are explained in the manual instalation section.
+
+## Manual instalation
+
+First, we need to add a couple of options to the scala compiler in the configuration section:
+
+```xml
+<plugin>
+<groupId>net.alchim31.maven</groupId>
+<artifactId>scala-maven-plugin</artifactId>
+<version>4.0.2</version>
+...
+<configuration>
+    <compilerPlugins>
+        <compilerPlugin>
+            <groupId>org.scalameta</groupId>
+            <artifactId>semanticdb-scalac_${scala.version}</artifactId>
+            <version>@@SCALAMETA_VERSION@@</version>
+        </compilerPlugin>
+    </compilerPlugins>
+    <args>
+        <arg>-P:semanticdb:synthetics:on</arg>
+        <arg>-P:semanticdb:failures:warning</arg>
+        <arg>-P:semanticdb:sourceroot:${project.basedir}</arg>
+        <arg>-Yrangepos</arg>
+        <arg>-Xplugin-require:semanticdb</arg>
+    </args>
+</configuration>
+...
+</plugin>
+```
+
+`@@SCALAMETA_VERSION@@` version can be replaced with whatever is the newest semanticDB version.
+
+Next, we need to do run bloopInstall via maven, which can be done easily through:
+
+`mvn ch.epfl.scala:maven-bloop_2.10:1.3.0:bloopInstall -DdownloadSources=true`
+
+where `1.3.0` can be replaced with newer bloop version.
+
+Everything should now be correctly configured and work even when reimporting the project.
+
+If you don't want to modify the `pom.xml` you can also run bloopInstall with an additional parameter:
+
+`mvn ch.epfl.scala:maven-bloop_2.10:1.3.0-RC1:bloopInstall -DdownloadSources=true -DaddScalacArgs=-Xplugin:@@semantic_DB_plugin_jar@@|-P:semanticdb:synthetics:on|-P:semanticdb:failures:warning|-P:semanticdb:sourceroot:/home/tgodzik/Documents/maven-test-repo|-Yrangepos|-Xplugin-require:semanticdb'`
```

Comment:
````suggestion
```
mvn ch.epfl.scala:maven-bloop_2.10:@@BLOOP_VERSION@@:bloopInstall -DdownloadSources=true -DaddScalacArgs=-Xplugin:/path/to/semanticdb-scalac.jar|-P:semanticdb:synthetics:on|-P:semanticdb:failures:warning|-P:semanticdb:sourceroot:/path/to/workspace|-Yrangepos|-Xplugin-require:semanticdb'
```
````